pkg/driver/microphone/microphone.go

```diff
@@ -87,6 +90,9 @@ func (m *microphone) Open() error {
 }
 
 func (m *microphone) Close() error {
+	if m.deviceCloseFunc != nil {
+		m.deviceCloseFunc()
+	}
 	if m.chunkChan != nil {
 		close(m.chunkChan)
```

Comment:
I think this can be called at exactly same timing with `m.chunkChan <- chunk` and cause panic by low chance.
(same as https://github.com/pion/mediadevices/pull/414#discussion_r930515146)

Reply:
I don't think it can be. The close function calls Stop/Uninit (turns out only Uninit is required). Both of those functions wait for the worker thread to return and ultimately will no longer call the stop/data callbacks. Therefore there are no more sends on the channel after and we can safely close it. Additionally you can see the Remarks section in https://miniaudio.docsforge.com/master/api/#ma_device_callback_proc referring to deadlocks and https://miniaudio.docsforge.com/master/api/ma_device_uninit/ showing the worker thread being waited on to terminate.

Reply:
Oh I see, as long as the uninit waits for ongoing callback return, it should be safe.

Maybe moving `close(m.chunkChan)` into `deviceCloseFunc` would be better at the readability about this safety logic.

Reply:
moved!